cmd/mt-whisper-importer-reader/main.go

```diff
@@ -0,0 +1,339 @@
+
+func (a pointSorter) Len() int           { return len(a) }
+func (a pointSorter) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a pointSorter) Less(i, j int) bool { return a[i].Timestamp < a[j].Timestamp }
+
+// the whisper archives are organized like a ringbuffer. since we need to
+// insert the points into the chunks in order we first need to sort them
+func sortPoints(points pointSorter) pointSorter {
+	sort.Sort(points)
+	return points
+}
+
+func getMetric(w *whisper.Whisper, file string) (*archive.Metric, error) {
+	if len(w.Header.Archives) == 0 {
+		return nil, errors.New(fmt.Sprintf("ERROR: Whisper file contains no archives: %q", file))
+	}
+
+	archives := make([]archive.Archive, 0, len(w.Header.Archives))
+	var chunkSpan uint32
+	var rowKey string
+	name := getMetricName(file)
+
+	// md gets generated from the first archive in the whisper file
+	md := getMetricData(name, int(w.Header.Archives[0].SecondsPerPoint))
+
+	for archiveIdx, archiveInfo := range w.Header.Archives {
+		if archiveIdx > 0 {
+			rowKey = api.AggMetricKey(
+				md.Id,
+				whisper.AggregationMethod(w.Header.Metadata.AggregationMethod).String(),
+				archiveInfo.SecondsPerPoint,
+			)
+		} else {
+			rowKey = md.Id
+		}
+
+		// only read archive if archiveIdx is in readArchives
+		if _, ok := readArchives[archiveIdx]; !ok && len(readArchives) > 0 {
+			continue
+		}
+
+		encodedChunks := make([]chunk.IterGen, 0)
+
+		if len(chunkSpans)-1 < archiveIdx {
+			// if we have more archives than chunk spans are specified, we simply use the last one
+			chunkSpan = chunkSpans[len(chunkSpans)-1]
+		} else {
+			chunkSpan = chunkSpans[archiveIdx]
+		}
+
+		points, err := w.DumpArchive(archiveIdx)
+		if err != nil {
+			return nil, errors.New(fmt.Sprintf("ERROR: Failed to read archive %d in %q, skipping: %q", archiveIdx, file, err))
+		}
+
+		var point whisper.Point
+		var t0, prevT0 uint32
+		var c *chunk.Chunk
+
+		for _, point = range sortPoints(points) {
+			if point.Timestamp == 0 {
+				continue
+			}
+
+			t0 = point.Timestamp - (point.Timestamp % chunkSpan)
+			if prevT0 == 0 {
+				c = chunk.New(t0)
+				prevT0 = t0
+			} else if prevT0 != t0 {
+				log(fmt.Sprintf("Mark chunk at t0 %d as finished", prevT0))
+				c.Finish()
+
+				encodedChunks = append(encodedChunks, *chunk.NewBareIterGen(c.Bytes(), c.T0, chunkSpan))
+
+				log(fmt.Sprintf("Create new chunk at t0 %d with chunk span %d", t0, chunkSpan))
+				c = chunk.New(t0)
+				prevT0 = t0
+			}
+
+			err := c.Push(point.Timestamp, point.Value)
+			if err != nil {
+				return nil, errors.New(fmt.Sprintf("ERROR: Failed to push value into chunk at t0 %d: %q", t0, err))
+			}
+		}
+
+		if int64(point.Timestamp) > md.Time {
+			md.Time = int64(point.Timestamp)
+		}
+
+		// if the last written point was also the last one of the current chunk,
+		// or if skipUnfinishedChunks is on, we close the chunk and
+		if point.Timestamp == t0+chunkSpan-archiveInfo.SecondsPerPoint || !*skipUnfinishedChunks {
```

Comment:
i don't like double negatives.  Can't we just call this `writeUnfinishedChunks` and default to false?